kernel/arch/dreamcast/hardware/maple/controller.c

```diff
@@ -25,9 +29,21 @@ typedef struct cont_cond {
     uint8_t joy2y;     /* second joystick Y */
 } cont_cond_t;
 
-static cont_btn_callback_t btn_callback = NULL;
-static uint8_t btn_callback_addr = 0;
-static uint32_t btn_callback_btns = 0;
+typedef struct cont_callback_params {
```

Comment:
The problem with that is that the entire user-facing API was written prefixing everything with `cont_` and `CONT_`, so this would be a whole breaking API change... Something more to note for 3.0 with the other API breaks we want to make, if that's what we want to do.